app/Post_Types/Archive_Item/Rest.php

```diff
@@ -0,0 +1,343 @@
+<?php
+
+namespace Diviner\Post_Types\Archive_Item;
+
+use Carbon_Fields\Field;
+use Diviner\Post_Types\Diviner_Field\Diviner_Field;
+use Diviner\Post_Types\Diviner_Field\PostMeta as FieldPostMeta;
+use Diviner\Post_Types\Diviner_Field\Types\Text_Field;
+use Diviner\Post_Types\Diviner_Field\Types\Date_Field;
+use Diviner\Post_Types\Diviner_Field\Types\Taxonomy_Field;
+use Diviner\Post_Types\Diviner_Field\Types\CPT_Field;
+use Diviner\Post_Types\Diviner_Field\Types\Select_Field;
+use Diviner\Admin\Settings;
+use Diviner\CarbonFields\Helper;
+
+
+class Rest {
+
+	const FIELD_INDEX_ID = 'id';
+	const FIELD_INDEX_TITLE = 'title';
+	const FIELD_INDEX_TYPE = 'type';
+	const FIELD_INDEX_FIELD_ID = 'field_id';
+	const FIELD_INDEX_FIELD_POPUP = 'field_popup';
+
+	protected $container;
+	protected $fields;
+
+	public function init() {
+		$this->fields = $this->get_fields();
+	}
+
+	public function hooks() {
+		$action = 'after_setup_theme';
+		if ( DIVINER_IS_PLUGIN ) {
+			$action = 'plugins_loaded';
+		}
+		// add_filter( $action, [$this, 'init'], 10, 2 );
+		add_filter( 'rest_' . Archive_Item::NAME . '_query', [$this, 'rest_api_filter_add_vars'], 10, 2 );
+		add_action( 'rest_api_init', array( &$this,'custom_register_rest_fields') );
+	}
+
+	protected function get_fields() {
+		if (isset($this->fields)) {
+			return $this->fields;
+		}
+		$meta_query = array(
+			array(
```

Comment:
For debug purposes: you can "name", by assigning a key to the `meta_query` entry, the meta_query sub-query you are running:
```php
$meta_query = array(
	'diviner_is_active` => array(...)
);
```

This will not be reflected in the slug but any dump or inspection will be immensely facilitated.
Furthermore merging different meta_queries will become more deterministic.

Reply:
very cool. done